stub-connector/src/test/java/uk/gov/ida/notification/stubconnector/MetricsAppRuleTests.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.ida.notification.stubconnector;
+
+import io.dropwizard.testing.ConfigOverride;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import keystore.KeyStoreResource;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.opensaml.core.config.InitializationService;
+import uk.gov.ida.notification.VerifySamlInitializer;
+import uk.gov.ida.notification.apprule.rules.MetadataClientRule;
+import uk.gov.ida.saml.core.test.TestEntityIds;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import static io.dropwizard.testing.ResourceHelpers.resourceFilePath;
+import static keystore.builders.KeyStoreResourceBuilder.aKeyStoreResource;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.METADATA_SIGNING_A_PUBLIC_CERT;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_PRIVATE_KEY;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_PUBLIC_CERT;
+import static uk.gov.ida.saml.core.test.builders.CertificateBuilder.aCertificate;
+
+public class MetricsAppRuleTests {
+
+    @ClassRule
+    public static final MetadataClientRule metadataClientRule;
+
+    private static final KeyStoreResource truststore = aKeyStoreResource()
+            .withCertificate("VERIFY-FEDERATION", aCertificate().withCertificate(METADATA_SIGNING_A_PUBLIC_CERT).build().getCertificate())
+            .build();
+
+    static {
+        try {
+            InitializationService.initialize();
+            VerifySamlInitializer.init();
+            metadataClientRule = new MetadataClientRule();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        truststore.create();
+    }
+
+    @Rule
+    public DropwizardAppRule<StubConnectorConfiguration> stubConnectorAppRule = new DropwizardAppRule<>(StubConnectorApplication.class, resourceFilePath("config.yml"),
```

Comment:
Why not pull this out into a `StubConnectorAppRule` class to match the other apps?

Reply:
Can do, but there's only one test right now :D